libqtile/group.py

```diff
@@ -243,31 +245,32 @@ def add(self, win, focus=True, force=False):
             win._float_state = FloatStates.FLOATING
         if win.floating:
             self.floating_layout.add(win)
-        else:
-            for i in self.layouts:
-                i.add(win)
+        for i in self.layouts:
+            i.add(win)
         if focus:
             self.focus(win, warp=True, force=force)
 
     def remove(self, win, force=False):
         self.windows.remove(win)
         hadfocus = self._remove_from_focus_history(win)
         win.group = None
+        nextfocus = None
+
+        for i in self.layouts:
+            if i is self.layout:
+                nextfocus = i.remove(win)
+            else:
+                i.remove(win)
 
         if win.floating:
             nextfocus = self.floating_layout.remove(win)
 
+        if win.floating:
             nextfocus = nextfocus or \
                 self.current_window or \
                 self.layout.focus_first() or \
                 self.floating_layout.focus_first(group=self)
```

Comment:
The new line 268 is redundant

Reply:
Thanks! I have seen a few inconsistencies here and there so I will fix them later today 😅

Reply:
Should be fixed now